flows/engine/assets.go

```diff
@@ -54,7 +55,8 @@ type assetsEnvelope struct {
 
 func ReadAssets(data json.RawMessage) (flows.Assets, error) {
 	var envelope assetsEnvelope
-	if err := json.Unmarshal(data, &envelope); err != nil {
+	err := utils.UnmarshalAndValidate(data, &envelope, "assets")
+	if err != nil {
```

Comment:
The former is more go-y, but I actually haven't gotten comfortable with it yet..